synchronizer/synchronizer.go

```diff
@@ -825,8 +818,14 @@ func (s *ClientSynchronizer) checkReorgAndExecuteReset(lastEthBlockSynced *state
 			log.Errorf("error resetting the state to a previous block. Retrying... Err: %s", err.Error())
 			return false, lastEthBlockSynced, fmt.Errorf("error resetting the state to a previous block. Err: %w", err)
 		}
-		return true, block, nil
+		newLastBlock, err := s.state.GetLastBlock(s.ctx, nil)
+		if err != nil {
+			log.Warnf("error getting last block synced from db, returning expected block %d. Error: %v", block.BlockNumber, err)
+			return false, block, err
+		}
+		return true, newLastBlock, nil
```

Comment:
I don't understand this change. I think it is redundant

Reply:
yes, it's redudant, but it a double-check that we after reset we got the expected block